src/Value/StoreId.php

```diff
@@ -1,51 +1,34 @@
 <?php
+declare(strict_types=1);
 
 namespace MyOnlineStore\Common\Domain\Value;
 
 use Doctrine\ORM\Mapping as ORM;
-use MyOnlineStore\Common\Domain\Assertion\NumericAssertionTrait;
 
 /**
  * @ORM\Embeddable
  */
 final class StoreId
 {
-    use NumericAssertionTrait;
-
     /**
      * @ORM\Column(name="store_id", type="integer")
      *
-     * @var int|string
+     * @var int
      */
     private $id;
 
-    /**
-     * @param int $id
-     */
-    public function __construct($id)
+    public function __construct(int $id)
```

Comment:
This change will generate a **lot** of work to get compatible within the monolith

Reply:
I do not intend to ever make this compatible within the monolith, but only for use in other services.

Reply:
That would mean we'd need to get very creative with major versions if we want to introduce something with BC to the monolith and forever maintain a separate package version with backports, even when it gets upgraded to new PHP versions